core/che-core-metrics-core/src/main/java/org/eclipse/che/core/metrics/ApiResponceMetricFilter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.core.metrics;
+
+import java.io.IOException;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+
+@Singleton
+public class ApiResponceMetricFilter implements Filter {
+
+  @Inject protected ApiResponseCounter apiResponseCounter;
```

Comment:
Other places in the codebase seem to prefer constructor injection with private final fields. Is it also possible for the servlet filters?

Reply:
I saw in other filters there is also method or field injection. And I had to do it when it come to running rest-assured tests that couldn't instantiate a filter that would have constructor with arguments

Reply:
:+1: thanks for the explanation. Maybe a comment at the field explaining this would help the casual observer in the future ;)